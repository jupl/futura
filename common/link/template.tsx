import NextLink, {LinkState} from 'next/link'
import * as React from 'react'

/** Inherited properties */
export type Parent = Pick<LinkState, Exclude<keyof LinkState, 'children'>>

/** Component properties */
export interface Props {
  children?: React.ReactNode
  className?: string
}

/**
 * Render link that is style-component friendly
 * @param props Component properties
 * @return Link
 */
export function Link({className, children, ...props}: Props & Parent) {
  return (
    <NextLink {...props}>
      <a children={children} className={className} />
    </NextLink>
  )
}
