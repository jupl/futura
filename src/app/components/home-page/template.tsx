import React from 'react'
import styled from 'styled-components'

/** Component properties */
export interface Props {
  ssr: boolean
}

// Container component
const Container = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  overflow: auto;
  display: flex;
  justify-content: center;
  align-items: center;
  background: linear-gradient(white, gainsboro);
`

/**
 * Home page component
 * @return Home page
 */
export const AppHomePage = ({ssr}: Props) => (
  <Container>
    Hello, Next.js from {ssr ? 'server' : 'client'}
  </Container>
)
