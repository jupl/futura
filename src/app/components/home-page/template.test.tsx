import {shallow} from 'enzyme'
import React from 'react'
import {AppHomePage} from './template'

test('<AppHomePage> template', () => {
  expect(shallow(<AppHomePage ssr />)).toMatchSnapshot()
  expect(shallow(<AppHomePage ssr={false} />)).toMatchSnapshot()
})
