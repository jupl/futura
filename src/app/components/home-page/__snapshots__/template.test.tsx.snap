// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHomePage> template 1`] = `
<styled.div>
  Hello, Next.js from 
  server
</styled.div>
`;

exports[`<AppHomePage> template 2`] = `
<styled.div>
  Hello, Next.js from 
  client
</styled.div>
`;
